### PR TITLE
address bugprone-exception-escape warning from clang-tidy.

### PR DESCRIPTION
**Pull request by fujitatomoya**
draft for https://github.com/ros2/rclcpp/issues/1890

Signed-off-by: Tomoya Fujita <Tomoya.Fujita@sony.com>

### CODE REVIEW

**Review by ivanpauno — APPROVED** [review]
LGTM!

Did you verify this solves https://github.com/ros2/rclcpp/issues/1890?

**Comment by fujitatomoya** [comment]
@ivanpauno no, im not sure how to do that... @alsora could i have some help here? or could you tell me how to verify?

File: rclcpp_components/src/node_main.cpp.in [hunk 1]
```diff
@@ -37,7 +37,15 @@ int main(int argc, char * argv[])
   std::string class_name = "rclcpp_components::NodeFactoryTemplate<@component@>";
 
   RCLCPP_DEBUG(logger, "Load library %s", library_name.c_str());
-  auto loader = new class_loader::ClassLoader(library_name);
+  try {
+    auto loader = new class_loader::ClassLoader(library_name);
```
Comment (alsora) [inline rclcpp_components/src/node_main.cpp.in:41]: this code does not compile.
you should have
`class_loader::ClassLoader loader = nullptr;` 
outside of the try catch

**Review by alsora — CHANGES_REQUESTED** [review]
The code does not compile as it is.
However, with a minimal change (described above) it does and the try catch block fixes the bugprone warning.

File: rclcpp_components/src/node_main.cpp.in (hunk 1 above)
Comment (alsora) [reply rclcpp_components/src/node_main.cpp.in:41]: slightly unrelated, but is there a reason why we use a raw pointer here?
this is not even deallocated at the end of the file (not a big deal, but still bad to see).

**Comment by alsora** [comment]
Sorry, I made a mistake. The error is still there even after your patch.
It looks like this is the function throwing an exception https://github.com/ros/class_loader/blob/galactic/include/class_loader/class_loader_core.hpp#L316

**Comment by alsora** [comment]
@fujitatomoya I opened a PR that fixes the issue: https://github.com/ros2/rclcpp/pull/1895

**Comment by fujitatomoya** [comment]
@alsora thanks! i will review https://github.com/ros2/rclcpp/pull/1895 instead.

**Pull request closed without merging**